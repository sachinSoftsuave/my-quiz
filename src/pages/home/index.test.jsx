import React from "react";
import { mount } from "enzyme";
import "jsdom-global/register";
import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Home from "./index.jsx";
import { act } from "react-test-renderer";
Enzyme.configure({ adapter: new Adapter() });
import toJson from "enzyme-to-json";

const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

const mockStartQuiz = jest.fn();

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useContext: () => ({
      setQuizStarted: () => mockStartQuiz,
      isQuizStarted: true,
  }),
}));

const prop = {
    handleStart:jest.fn()
}


describe("<Home/>", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<Home />);

  });
  it("it should render title", () => {
    expect(wrapper.find("h1").text()).toContain("Coding Quiz Challenge");
  });

  it("it should render subtitle1", () => {
    expect(wrapper.find(".subTitleText1").text()).toContain(
      "Try to answer to code-related question within the timelimit"
    );
  });

  it("it should render subtitle2", () => {
    expect(wrapper.find(".subTitleText2").text()).toContain(
      "Keep in mind that incorrect answers will penalize your score/time by ten seconds"
    );
  });

  it("should render the start button", () => {
    expect(wrapper.find("button").text()).toContain("Start Quiz");
  });


   it('Test click event', async () => {
    await act(async () => {
    await wrapper.find('button').simulate('click');
    })
    console.log(window.location.pathname,"myurl");
  });

  it('it renders the snapshot of home component', () =>{
       expect(toJson(wrapper)).toMatchSnapshot();
  })
});
