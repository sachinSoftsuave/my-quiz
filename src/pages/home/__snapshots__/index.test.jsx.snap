// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> it renders the snapshot of home component 1`] = `
<Home>
  <div
    className="homeContainer"
  >
    <div
      className="modelHome"
    >
      <h1
        className="titleText"
      >
        Coding Quiz Challenge
      </h1>
      <p
        className="subTitleText1"
      >
        Try to answer to code-related question within the timelimit
      </p>
      <p
        className="subTitleText2"
      >
        Keep in mind that incorrect answers will penalize your score/time by ten seconds
      </p>
      <div
        className="startBtnContainer"
      >
        <button
          className="startBtn"
          onClick={[Function]}
        >
          Start Quiz
        </button>
      </div>
    </div>
  </div>
</Home>
`;
